src/screens/Forum/ForumHomeScreen.js

```diff
@@ -26,56 +35,138 @@ export default class ForumHomeScreen extends React.Component {
       }
     });
 
-    this.unsubscribeFromFirestore = firestore().collection('forum_posts')
+    const { postLimit } = this.state;
+    firestore().collection('forum_posts')
       .orderBy('createdAt', 'desc')
-      .onSnapshot((snapshot) => {
+      .limit(postLimit)
+      .get()
+      .then((snapshot) => {
         // Store data for posts in state
         const forumPosts = snapshot.docs.map((doc) => ({ ...doc.data(), id: doc.id }));
-        this.setState({ forumPosts, loading: false });
-      }, (error) => {
-        this.setState({ errorMessage: error.message, loading: false });
-      });
+        const lastReferencedPost = forumPosts[forumPosts.length - 1];
+
+        this.setState({ forumPosts, lastReferencedPost, loading: false });
+      })
+      .catch((error) => this.setState({ errorMessage: error.message, loading: false }));
   }
 
   componentWillUnmount() {
     this.unsubscribeFromAuth();
-    this.unsubscribeFromFirestore();
   }
 
   navigateToPostScreen(postID, userID) {
     const { navigation } = this.props;
     navigation.navigate('ForumPost', { postID, userID });
   }
 
+  handleEndReached() {
+    const { allPostsLoaded } = this.state;
+
+    if (!allPostsLoaded) {
+      this.loadMore();
+    }
+  }
+
+  // Fetch more data from firestore to load next posts
+  loadMore() {
+    this.setState({ loadingMore: true });
+    const { postLimit, forumPosts, lastReferencedPost } = this.state;
+
+    firestore().collection('forum_posts')
+      .orderBy('createdAt', 'desc')
+      .startAfter(lastReferencedPost.createdAt)
+      .limit(postLimit)
+      .get()
+      .then((snapshot) => {
+        const newForumPosts = snapshot.docs.map((doc) => ({ ...doc.data(), id: doc.id }));
+        const newLastReferenced = newForumPosts[newForumPosts.length - 1];
+
+        if (newForumPosts.length === 0) {
+          this.setState({ allPostsLoaded: true, loadingMore: false });
+        } else {
+          this.setState({ // Store/append updated data for next postLimit# of posts in state
+            forumPosts: [...forumPosts, ...newForumPosts],
+            lastReferencedPost: newLastReferenced,
+          });
+        }
+      })
+      .catch((error) => this.setState({ errorMessage: error.message, loadingMore: false }));
+  }
+
+  renderHeaderComponent() {
+    const { loading, errorMessage } = this.state;
+
+    if (loading && !errorMessage) {
+      return (
+        <View style={styles.activityIndicator}>
+          <ActivityIndicator />
+        </View>
+      );
+    }
+    if (errorMessage && !loading) {
+      return (
+        <Text style={{ color: 'red' }}>
+          {errorMessage}
+        </Text>
+      );
+    }
+    if (errorMessage && loading) {
+      return (
+        <View>
```

Comment:
Would it be possible to simplify these three statement into one by using inline ternary statements?

Reply:
(from previous pr) These were initially inline ternary statements, but that format caused some configuration issues so they were changed to this format instead.

Reply:
Alright, sounds good!